lib/pushy_client/protocol_handler.rb

```diff
@@ -476,6 +468,35 @@ def send_signed_json_command(method, json)
       end
     end
 
+    # This method validate and generate the message
+    # params user packet
+    # Return Waring or JSON message
+    def validate_and_generate_json(json)
+      message = generate_json(json)
+      unless is_valid(message)
+        deleted_packet = json.select { |k| [:stdout, :stderr].include? k }
+        json.delete_if { |k| [:stdout, :stderr].include? k }
+        Chef::Log.warn("Deleted packet #{deleted_packet} from Client packet") if deleted_packet.to_s.size > 1
+        # Re-generate json message
+        message = generate_json(json)
+      end
+      unless is_valid(message)
+        Chef::Log.warn("Client message size #{message.to_s.bytesize} is larger than the Client MAX_BODY_SIZE of #{client.max_body_size.to_i}")
+      else
+        message
+      end
+    end
+
+    # This method generate json
+    def generate_json(json)
+      JSON.generate(json)
+    end
```

Comment:
Do we really need to define another method to generate json as JSON is already had this method defined ?